src/output-json-alert.c

```diff
@@ -226,7 +249,11 @@ void AlertJsonHeader(void *ctx, const Packet *p, const PacketAlert *pa, JsonBuil
         AlertJsonSourceTarget(p, pa, js, addr);
     }
 
-    if ((json_output_ctx != NULL) && (flags & LOG_JSON_RULE_METADATA)) {
```

Comment:
I wonder if there are more unneeded NULL checks we could find automatically

Reply:
Possibly but perhaps as part of a special purpose ticket to address that?

Reply:
I surely not meant to do it within this PR :-)